redo.dev/src/components/toolSummaries.tsx

```diff
@@ -133,7 +145,17 @@ const DemoButton = (props: ToolSummaryProps) => {
             variant="contained"
             sx={{ whiteSpace: "nowrap" }}
             onClick={() => {
+                setTimeout(() => {
+                    props.setLoading(false)
+                }, 3000)
+                props.setLoading(isActiveDemo ? false : true)
                 props.setActiveDemo(isActiveDemo ? null : props.demoElement!)
+                setTimeout(() => {
+                    props.demoRef.current!.scrollIntoView({
```

Comment:
This should probably be:

`props.demoRef.current?.scrollIntoView(...)`

This will avoid the TS error and will just do nothing if current is not set.